ocaml/xapi/xapi_vm.ml

```diff
@@ -256,14 +256,36 @@ let start ~__context ~vm ~start_paused ~force =
   in
   let vmr =
     if platform <> vmr.API.vM_platform then begin
-      Db.VM.set_platform ~__context ~self:vm ~value:platform;
       { vmr with API.vM_platform = platform }
     end else vmr
   in
 
+  (* Check to see if secure boot should be parsed *)
+  let pool = Helpers.get_pool ~__context in
+  if List.assoc "secureboot" platform = "auto" then
+    Db.VM.set_platform ~__context ~self:vm
+      ~value:(("secureboot", string_of_bool (Db.Pool.get_uefi_certificates ~__context ~self:pool <> "")) :: List.remove_assoc "secureboot" platform)
+  else
+    Db.VM.set_platform ~__context ~self:vm ~value:platform;
 
-  (* Check to see if we're using correct device-model when vm has VUSBs*)
+  let dir = ref [||] in
+  try
+    dir := Sys.readdir "/usr/share/varstored";
+  with Sys_error _ ->
+    begin
+      debug "varstored directory does not exist";
+      Xapi_stdext_unix.Unixext.mkdir_rec "/usr/share/varstored" 0o755
```

Comment:
The path should probably go into Xapi_globs, to allow overriding it. An entry should be added to nonessential_dirs.